pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/mergerollup/MergeRollupTaskGenerator.java

```diff
@@ -656,18 +666,25 @@ private List<PinotTaskConfig> createPinotTaskConfigs(List<SegmentZKMetadata> sel
     int numRecordsPerTask = 0;
     List<List<String>> segmentNamesList = new ArrayList<>();
     List<List<String>> downloadURLsList = new ArrayList<>();
+    List<SegmentZKMetadata> segments = new ArrayList<>();
     List<String> segmentNames = new ArrayList<>();
     List<String> downloadURLs = new ArrayList<>();
 
     for (int i = 0; i < selectedSegments.size(); i++) {
       SegmentZKMetadata targetSegment = selectedSegments.get(i);
+      segments.add(targetSegment);
       segmentNames.add(targetSegment.getSegmentName());
       downloadURLs.add(targetSegment.getDownloadUrl());
       numRecordsPerTask += targetSegment.getTotalDocs();
       if (numRecordsPerTask >= maxNumRecordsPerTask || i == selectedSegments.size() - 1) {
-        segmentNamesList.add(segmentNames);
-        downloadURLsList.add(downloadURLs);
+        // Skip generating rollup task with all merged segments
+        if (MergeTaskUtils.getMergeType(mergeConfigs) != MergeType.ROLLUP
```

Comment:
Is there any reason for wanting to skip tasks specifically for Rollup tasks? Should this apply to concat case as well?

Reply:
I think it can be applied to concat an dedup as well. Let me update.